src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2871,4 +2873,38 @@ private static boolean validateIpAddress(String string) {
         }
         return true;
     }
+
+    private static final class StandardDataStream implements WriteListener {
+        private static final Logger logger = LoggerFactory.getLogger(
+                StandardDataStream.class);
+
+        private final AsyncContext async;
+        private final InputStream is;
+        private final ServletOutputStream sos;
+
+        private StandardDataStream(AsyncContext async, InputStream is,
+                ServletOutputStream sos) {
+            this.async = async;
+            this.is = is;
+            this.sos = sos;
+        }
+
+        public void onWritePossible() throws IOException {
+            byte[] buffer = new byte[4096];
+            while (sos.isReady()) {
+                int len = is.read(buffer);
```

Comment:
@andrewgaul this won't solve the problem we are addressing. This will just write the response to the client in an asynchronous fashion. Even in this case is.read(buffer) is a blocking call. What we wanted to solve is when we make a request to put blob or get blob we will return a listenable future which will be completed only when there is a response from the service provider.